packages/spark/sodasql/dialects/spark_dialect.py

```diff
@@ -9,31 +9,175 @@
 from sodasql.scan.dialect import Dialect, SPARK, KEY_WAREHOUSE_TYPE
 from sodasql.scan.parser import Parser
 
 logger = logging.getLogger(__name__)
 
 
+def hive_connection_function(
+    username: str,
+    password: str,
+    host: str,
+    port: str,
+    database: str,
+    auth_method: str,
+    **kwargs,
+) -> hive.Connection:
+    """
+    Connect to hive.
+
+    Parameters
+    ----------
+    username : str
+        The user name
+    password : str
+        The password
+    host: str
+        The host.
+    port : str
+        The port
+    database : str
+        The databse
+    auth_method : str
+        The authentication method
+
+    Returns
+    -------
+    out : hive.Connection
+        The hive connection
+    """
+    connection = hive.connect(
+        username=username,
+        password=password,
+        host=host,
+        port=port,
+        database=database,
+        auth=auth_method
+    )
+    return connection
+
+
+def _build_odbc_connnection_string(**kwargs: Any) -> str:
+    return ";".join([f"{k}={v}" for k, v in kwargs.items()])
+
+
+def decode_sketchy_utf8(raw_bytes: bytes) -> bytes:
+    """
+    Decode sketch utf8 characters.
+
+    Parameters
+    ----------
+    raw_bytes : bytes
+        The bytest to handle
+
+    Returns
+    -------
+    out : bytes
+        The handled bytes.
+
+    Source
+    ------
+    https://github.com/mkleehammer/pyodbc/issues/328
+    """
+    null_terminated_bytes = raw_bytes.split(b'\x00')[0]
+    return null_terminated_bytes.decode('utf-8')
+
+
+def odbc_connection_function(
+    driver: str,
+    host: str,
+    port: str,
+    token: str,
+    organization: str,
+    cluster: str,
+    server_side_parameters: Dict[str, str],
+    **kwargs,
+) -> pyodbc.Connection:
+    """
+    Connect to hive.
+
+    Parameters
+    ----------
+    driver : str
+        The path to the driver
+    host: str
+        The host.
+    port : str
+        The port
+    token : str
+        The login token
+    organization : str
+        The organization
+    cluster : str
+        The cluster
+    server_side_parameters : Dict[str]
+        The server side parameters
+
+    Returns
+    -------
+    out : pyobv.Connection
+        The connection
+    """
+    http_path = f"/sql/protocolv1/o/{organization}/{cluster}"
+    user_agent_entry = f"soda-sql-spark/{SODA_SQL_VERSION} (Databricks)"
+
+    connection_str = _build_odbc_connnection_string(
+        DRIVER=driver,
+        HOST=host,
+        PORT=port,
+        UID="token",
+        PWD=token,
+        HTTPPath=http_path,
+        AuthMech=3,
+        SparkServerType=3,
+        ThriftTransport=2,
+        SSL=1,
+        UserAgentEntry=user_agent_entry,
+        LCaseSspKeyName=0 if server_side_parameters else 1,
+        **server_side_parameters,
+    )
+    connection = pyodbc.connect(connection_str, autocommit=True)
+    connection.add_output_converter(pyodbc.SQL_VARCHAR, decode_sketchy_utf8)
```

Comment:
I am getting the following error when this is not set: 

```
UnicodeDecodeError: 'utf-16-le' codec can't decode bytes in position 718-719: illegal UTF-16 surrogate
```

See the github discussion in the source string where I found this solution. 

I want to think about this for a moment to decide if we should use this solution

Reply:
This issue is resolved by adding an aliases to the regexes